scripts/core/editor3/reducers/spellchecker.jsx

```diff
@@ -28,28 +24,47 @@ const spellchecker = (state = {}, action) => {
 const replaceWord = (state, text) => {
     const {editorState, spellcheckerMenu} = state;
     const {word, editorSelection} = spellcheckerMenu;
-
-    var textSelection;
-    var newContentState;
-    var newEditorState;
-
-    textSelection = editorSelection.merge({
+    const sel = editorSelection.merge({
         anchorOffset: word.offset,
         focusOffset: word.offset + word.text.length
     });
 
-    newContentState = Modifier.replaceText(editorState.getCurrentContent(), textSelection, text);
-    newEditorState = EditorState.push(editorState, newContentState, 'spellchecker');
+    var newState = Modifier.replaceText(editorState.getCurrentContent(), sel, text);
 
-    return Object.assign({}, state, {
-        editorState: newEditorState,
-        spellcheckerMenu: null
-    });
+    newState = EditorState.push(editorState, newState, 'spellchecker');
+
+    return {
+        ...state,
+        editorState: newState,
+        spellcheckerMenu: {
+            ...spellcheckerMenu,
+            visible: false
+        }
+    };
 };
 
 /**
  * @ngdoc method
- * @name Spellchecker Reducers#showContextMenu
+ * @name closeContextMenu
+ * @param {Object} state
+ * @return {Object} returns new state
+ * @description Save on store all relevant data for spellchecker context menu
+ */
+const closeContextMenu = (state) => {
```

Comment:
I think you mean for the context menu to manage its own state here, which is a bit tricky given that we use Redux. The context menu's visibility has to be part of the store because it is triggered by external components (like the decorator). In this case, the state of the editor remains unaffected, only the state of the spellchecker's visibility changes.

I don't see a problem with this implementation, and I also can't think of a different solution, but if you have a suggestion I am open to it!